critiquebrainz/db/review.py

```diff
@@ -0,0 +1,587 @@
+            "timestamp": review.pop("timestamp"),
+            "text": review.get("text"),
+            "review_id": review.get("id"),
+        }
+        review["user"] = User({
+            "id": review["user_id"],
+            "display_name": review.pop("display_name", None),
+            "is_blocked": review.pop("is_blocked", False),
+            "show_gravatar": review.pop("show_gravatar", False),
+            "musicbrainz_username": review.pop("musicbrainz_id"),
+            "email": review.pop("email"),
+            "created": review.pop("user_created"),
+        })
+        review["license"] = {
+            "id": review["license_id"],
+            "info_url": review["info_url"],
+            "full_name": review["full_name"],
+        }
+        votes = db_revision.votes(review["last_revision"]["id"])
+        review["votes_positive_count"] = votes["positive"]
+        review["votes_negative_count"] = votes["negative"]
+        review["rating"] = review["votes_positive_count"] - review["votes_negative_count"]
+    return review
+
+
+def set_hidden_state(review_id, *, is_hidden):
+    """Hide or unhide a review with a given ID.
+
+    Args:
+        review_id(uuid): ID of the review to unhide.
+        is_hidden(bool): True is review to be hidden. False if not.
+    """
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            UPDATE review
+               SET is_hidden = :is_hidden
+             WHERE id = :review_id
+        """), {
+            "review_id": review_id,
+            "is_hidden": is_hidden,
+        })
+
+
+def get_count(*, is_draft=False, is_hidden=False):
+    """Get number of reviews in CritiqueBrainz.
+
+    Args:
+        is_draft (bool): True if drafted reviews are to be counted.
+        is_hidden (bool): True if hidden reviews are to be counted.
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT count(*)
+              FROM review
+             WHERE is_draft = :is_drafted
+               AND is_hidden = :is_hidden
+        """), {
+            "is_drafted": is_draft,
+            "is_hidden": is_hidden,
+        })
+        return result.fetchone()[0]
+
+
+def update(*, review_id, drafted, text, license_id=None,
+           language=None, is_draft=None):
+    """Update contents of a review.
+
+    Args:
+        review_id (uuid): ID of the review.
+        drafted (bool): Whether review is presently drafted.
+        license_id(optional, str): Updated license id of a review.
+        is_draft(optional, bool): Whether to publish review or keep it drafted.
+        text(optional, str): Updated text of a review.
+    """
+    updates = []
+    updated_info = {}
+
+    if license_id is not None:
+        if not drafted: # If trying to convert published review into draft.
+            raise BadRequest(lazy_gettext("Changing license of a published review is not allowed."))
```

Comment:
One other important thing that I missed. We should try to avoid making Flask views and database access code closely tied. That means not using exceptions and functions from Flask.

Modules in the `critiquebrainz.db` package should raise exceptions from `critiquebrianz.db.exceptions`. I understand that this is not the old implementation, but that's the perfect time to make this improvement. This would mean we could easily use DB access code without Flask.